WordPress/gutenberg
# Mobile - Update editor initial HTML test for iOS and update Appium

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5197



## What?

It updates the initial HTML test for iOS using the `getBlockAtPosition` helper instead of manually adding retry logic when trying to locate the last block.

## Why?

There were quite a few failures lately related to the iOS test.

## How?


First, it updates the test to use `getBlockAtPosition` which has the latest code improvements to find a block in E2E tests.

It also updates:
- Appium to `1.22.3`
- iOS simulator to `iPhone 13`
- iOS to `15.4`

For Android:
- For CircleCI it updates the platform to `11.0`
- And the Appium version to: `1.22.1` (It doesn't support `1.22.3` yet)
- For the Github action, I couldn't get to update the current Android version `28`. I tried different approaches but it would fail to boot the emulator, it looks like the machine it uses it's too slow and it times out.

Dependabot updates that got merged before this PR:
- https://github.com/WordPress/gutenberg/pull/44748
- https://github.com/WordPress/gutenberg/pull/44744
- https://github.com/WordPress/gutenberg/pull/44743
- https://github.com/WordPress/gutenberg/pull/44741
- https://github.com/WordPress/gutenberg/pull/44749

## Testing Instructions
Circle CI should pass, maybe try restarting the jobs to see if they succeed again.

Gutenberg Mobile PRs should also pass, including full test checks.

## Screenshots or screencast 
N/A
